core/frontend/src/components/app/SettingsMenu.vue

```diff
@@ -264,24 +308,54 @@ export default Vue.extend({
       this.operation_in_progress = false
     },
     async remove_service_log_files(): Promise<void> {
-      this.prepare_operation('Removing system log files...')
+      this.deletion_log_abort_controller = axios.CancelToken.source()
+      this.deletion_in_progress = true
+      this.current_deletion_path = '...'
+      this.current_deletion_size = 0
+      this.current_deletion_total_size = 0
+      this.current_deletion_status = 'Starting deletion...'
 
-      await back_axios({
-        url: `${API_URL}/services/remove_log`,
-        method: 'post',
-        params: {
-          i_know_what_i_am_doing: true,
-        },
-        timeout: 20000,
-      })
-        .then(() => {
-          this.get_log_folder_size()
-        })
-        .catch((error) => {
-          this.operation_error = String(error)
-          notifier.pushBackError('REMOVE_SERVICES_LOG_FAIL', error)
+      try {
+        await back_axios({
+          url: `${API_URL}/services/remove_log_stream`,
+          method: 'post',
+          params: {
+            i_know_what_i_am_doing: true,
+          },
+          responseType: 'text',
+          onDownloadProgress: (progressEvent) => {
+            let result = parseStreamingResponse(progressEvent.currentTarget.response)
+            result = result.filter((fragment) => fragment.fragment !== -1)
+            const last_fragment = result.last()
+            const total_deleted = result
+              .reduce((acc, fragment) => acc + (JSON.parse(fragment?.data ?? '{}')?.size ?? 0), 0)
+
+            if (last_fragment?.data) {
+              try {
+                const info = JSON.parse(last_fragment.data)
+                this.current_deletion_path = info.path.length > 30 ? `...${info.path.slice(-20)}` : info.path
+                this.current_deletion_size = info.size
+                this.current_deletion_total_size = total_deleted
+                this.current_deletion_status = info.success ? 'Successfully deleted' : 'Failed to delete'
+              } catch (e) {
+                console.error('Error parsing deletion info:', e)
+              }
```

Comment:
Shouldn't the `current_deletion_status` be `Deleting...`, and then use a `finally()` to set it to `Sucessifully deleted`? It looks weird/confusing to see `Successfully deleted` while it's still deleting files.